srcCpp/ParameterManager.cxx

```diff
@@ -0,0 +1,945 @@
+    secureCertAuthority->set_description("Certificate authority file <optional>.\nDefault: \"./resource/secure/cacert.pem\"");
+    secureCertAuthority->set_type(T_STR);
+    secureCertAuthority->set_extra_argument(YES);
+    secureCertAuthority->set_group(SECURE);
+    parameterList["secureCertAuthority"] = AnyParameter(secureCertAuthority);
+
+    // TODO assigned a value if it is pub or sub
+    Parameter<std::string> *secureCertFile = new Parameter<std::string>();
+    secureCertFile->set_command_line_argument(std::make_pair("-secureCertFile","<file>"));
+    secureCertFile->set_description("Certificate file <optional>.\nDefault: \"./resource/secure/sub.pem\"");
+    secureCertFile->set_type(T_STR);
+    secureCertFile->set_extra_argument(YES);
+    secureCertFile->set_group(SECURE);
+    parameterList["secureCertFile"] = AnyParameter(secureCertFile);
+
+    Parameter<std::string> *securePrivateKey = new Parameter<std::string>();
+    securePrivateKey->set_command_line_argument(std::make_pair("-securePrivateKey","<file>"));
+    securePrivateKey->set_description("Private key file <optional>.\nDefault: \"./resource/secure/subkey.pem\"");
+    securePrivateKey->set_type(T_STR);
+    securePrivateKey->set_extra_argument(YES);
+    securePrivateKey->set_group(SECURE);
+    parameterList["securePrivateKey"] = AnyParameter(securePrivateKey);
+
+    Parameter<std::string> *secureLibrary = new Parameter<std::string>();
+    secureLibrary->set_command_line_argument(std::make_pair("-secureLibrary","<file>"));
+    secureLibrary->set_description("Private key file <optional>.\nDefault: \"./resource/secure/subkey.pem\"");
+    secureLibrary->set_type(T_STR);
+    secureLibrary->set_extra_argument(YES);
+    secureLibrary->set_group(SECURE);
+    parameterList["secureLibrary"] = AnyParameter(secureLibrary);
+
+    Parameter<unsigned long long> *secureDebug = new Parameter<unsigned long long>(1);
+    secureDebug->set_command_line_argument(std::make_pair("-secureDebug","<level>"));
+    secureDebug->set_type(T_NUMERIC);
+    secureDebug->set_extra_argument(YES);
+    secureDebug->set_range(-1, ULLONG_MAX);
+    secureDebug->set_group(SECURE);
+    secureDebug->set_internal(true);
+    parameterList["secureDebug"] = AnyParameter(secureDebug);
+
+#endif
+}
+
+
+// Parse the command line parameters and set the value
+bool ParameterManager::parse(int argc, char *argv[])
+{
+    unsigned long long var;
+    bool success = true;
+    // Copy all arguments into a container of strings
+    std::vector<std::string> allArgs(argv, argv + argc);
+
+    std::map<std::string, AnyParameter>::iterator it;
+    for (unsigned int i = 1; i < allArgs.size(); i++) {
+        for (it = parameterList.begin(); it != parameterList.end(); it++) {
+            if (IS_OPTION(allArgs[i].c_str(), it->second.get()->get_command_line_argument().first.c_str())) {
+                // NumArguments == 0
+                if (it->second.get()->get_extra_argument() == NO) {
```

Comment:
Save `it->second.get()` in a local variable.

Reply:
Done